mcf-browser/src/back-end/parse.ts

```diff
@@ -0,0 +1,239 @@
+  const allSeries = Object.keys(newData);
+
+  for (const series of allSeries) {
+    datapoints[series] = datapoints[series] ? datapoints[series] : {};
+    datapoints[series] = {...datapoints[series], ...newData[series]};
+  }
+
+  return datapoints;
+}
+
+/** Parse files and get the local nodes
+  * @param {Array<Blob>} fileList the list of blobs to be parsed
+  * @return {ParseFileResponse} an object containing all of the ids of the
+  *    subject nodes and the error messages
+  */
+async function getNodes(fileList: Blob[]) : Promise<ParseFileResponse> {
+  const finalReturn: ParseFileResponse = {
+    errMsgs: [],
+    localNodes: [],
+    datapoints: {},
+  };
+
+  // Find TMCF file, if it exists
+  let tmcfFile = null;
+  for (const file of fileList) {
+    const fileName = (file as File).name;
+    const fileExt = fileName.split('.').pop();
+
+    if (fileExt === 'tmcf') {
+      if (tmcfFile) {
+        // If another TMCF file was found, throw an error
+        finalReturn['errMsgs'] = finalReturn['errMsgs'].concat([{
+          'file': (tmcfFile as File).name,
+          'errs': [
+            ['-1', '', ERROR_MESSAGES.MULTIPLE_TMCF],
+          ],
+        }]);
+      }
+
+      tmcfFile = file;
+    }
+  }
+
+
+  for (const file of fileList) {
+    const fileName = (file as File).name;
+    const fileExt = fileName.split('.').pop();
+
+    if (fileExt === 'mcf') {
+      const mcfOut = await ParseMcf.readFile(file);
+
+      if (mcfOut['errMsgs'].length !== 0) {
+        finalReturn['errMsgs'] = finalReturn['errMsgs'].concat([
+          {
+            file: fileName,
+            errs: mcfOut['errMsgs'],
+          },
+        ]);
+      }
+
+      finalReturn['localNodes'] = finalReturn['localNodes'].concat(
+          mcfOut['localNodes'],
+      );
+    } else if (fileExt === 'csv') {
+      if (tmcfFile) {
+        const tmcfFileName = (tmcfFile as File).name;
+        const tmcfOut = await ParseTmcf.generateMcf(tmcfFile, file).then(
+            (mcf) => {
+              const mcfParser = new ParseMcf(
+                  tmcfFileName + '&' + fileName,
+              );
+              return mcfParser.parseMcfStr(mcf as string);
+            },
+        );
+
+        const datapoints = await ParseTmcf.generateDataPoints(tmcfFile, file);
+        finalReturn['datapoints'] = mergeDataPoints(
+            finalReturn['datapoints'],
+            datapoints,
+        );
+
+        if (tmcfOut['errMsgs'].length !== 0) {
+          finalReturn['errMsgs'] = finalReturn['errMsgs'].concat({
+            file: tmcfFileName,
+            errs: tmcfOut['errMsgs'],
+          });
+        }
+        finalReturn['localNodes'] = finalReturn['localNodes'].concat(
+            tmcfOut['localNodes'],
+        );
+      }
+    }
+  }
+  return finalReturn;
+}
+
+/** Group nodes and find all time series
+  * @param {TimeDataObject} datapoints the time series data
+  * @return {Series[]} an array of time series in the data
```

Comment:
this type doesn't seem to match the actual returned type

Reply:
updated the docstring!